edb/server/protocol/binary.pyx

```diff
@@ -1341,135 +1071,37 @@ cdef class EdgeConnection:
-
-                if query_unit.tx_savepoint_rollback:
-                    _dbview.rollback_tx_to_savepoint(query_unit.sp_name)
-
-                if query_unit.tx_savepoint_declare:
-                    _dbview.declare_savepoint(
-                        query_unit.sp_name, query_unit.sp_id)
-
-                if query_unit.create_db:
-                    await self.server.introspect_db(
-                        query_unit.create_db
-                    )
-
-                if query_unit.drop_db:
-                    self.server._on_after_drop_db(
-                        query_unit.drop_db)
-
-                if config_ops:
-                    await _dbview.apply_config_ops(
-                        conn,
-                        config_ops)
-        except Exception as ex:
-            _dbview.on_error()
-
-            if (
-                query_unit.tx_commit and
-                not conn.in_tx() and
-                _dbview.in_tx()
-            ):
-                # The COMMIT command has failed. Our Postgres connection
-                # isn't in a transaction anymore. Abort the transaction
-                # in dbview.
-                _dbview.abort_tx()
-            raise
-        else:
-            side_effects = _dbview.on_success(query_unit, new_types)
-            if side_effects:
-                self.signal_side_effects(side_effects)
-            if not _dbview.in_tx():
-                state = _dbview.serialize_state()
-                if state is not orig_state:
-                    # In 3 cases the state is changed:
-                    #   1. The non-tx query changed the state
-                    #   2. The state is synced with dbview (orig_state is None)
-                    #   3. We came out from a transaction (orig_state is None)
-                    conn.last_state = state
+            await execute.execute(
+                conn,
+                dbv,
+                compiled,
+                bind_args,
+                fe_conn=self,
+                use_prep_stmt=use_prep_stmt,
+            )
         finally:
             self.maybe_release_pgcon(conn)
 
-    async def _finish_set_global(self, conn, query_unit, state):
-        config_ops = None
-        try:
-            try:
-                if state is not None:
-                    await self.wait_for_state_resp(
-                        state, bool(query_unit.tx_id))
-                for sql in query_unit.sql:
-                    data = await conn.wait_for_command(
-                        ignore_data=False
-                    )
-                if data:
-                    config_ops = [
-                        config.Operation.from_json(r[0][1:])
-                        for r in data]
-            finally:
-                await conn.wait_for_sync()
-        finally:
-            await conn.after_command()
-        return config_ops
+        query_unit = compiled.query_unit_group[0]
+        if query_unit.system_config:
```

Comment:
This was `if query_unit.config_requires_restart:`, is it a typo here to use `system_config`?

Reply:
Fixed in https://github.com/edgedb/edgedb/pull/5250/commits/050316468741e32340df7e2c7c297885d1fa07fd